application/core/plugins/dateFunctions/EMFunctions.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * This file is part of dateFunctions plugin
+ * @version 1.0.0
+ */
+
+namespace dateFunctions;
+
+use LimeExpressionManager;
+use Survey;
+
+class EMFunctions
+{
+    /**
+     * Formats a date according to the Survey's date format for the specified language
+     * @param string $date : a date in "Y-m-d H:i:s" format. Example: VALIDFROM.
+     * @param string|null $language : the language used for localization. Defaults to current session language. If the language is not configured in the survey, the base language will be used. Example: TOKEN:LANGUAGE
+     * @return string
+     */
+    public static function localize_date($date, $language = null)
+    {
+        if (empty($date)) {
+            return '';
+        }
+
+        // Try to get Survey ID from the EM
+        $surveyId = LimeExpressionManager::getLEMsurveyId();
```

Comment:
when called from email, this variable is not set on EM. That;s why we keep on trying for other alternatives if no SID found.

Reply:
Maybe we can create a Dev mantis issue for this.

It set in `\Yii::app()->session['LEMsid'] ` then seems OK for me.